serverless/serverless
# Improve Stage and Region Usage

## What did you implement:


Centralize throughout the codebase to using the `aws` provider's `getStage` and `getRegion` methods.

Remove the errant (but understandable) distributed usage of region and stage settings.  This otherwise locks in a multitude of bugs around the improper algorithm for selecting (given all context) the proper region or stage setting.  Instead, all code should use the centralized algorithm for determining such values.  This creates a strange first and second class configuration concept but these two are sufficiently varied and complex in their creation and use that this seems appropriate.



## How did you implement it:



Centralize throughout the codebase to using the `aws` provider's `getStage` and `getRegion` methods.

## How can we verify it:



verify by noting the tests continue to pass.

is there something more extensive than the unit tests that I should run in the future to verify across the cli interactions or the like?

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
